library/kotlin/src/io/envoyproxy/envoymobile/GRPCStreamEmitter.kt

```diff
@@ -0,0 +1,31 @@
+package io.envoyproxy.envoymobile
+
+import java.nio.ByteBuffer
+
+
+class GRPCStreamEmitter(
+    private val emitter: StreamEmitter
+) {
+  fun sendMessage(message: ByteBuffer) {
+    // https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests
+    // Length-Prefixed-Message = Compressed-Flag | Message-Length | Message
+    // Compressed-Flag = 0 / 1, encoded as 1 byte unsigned integer
+    // Message-Length = length of Message, encoded as 4 byte unsigned integer (big endian)
+    // Message = binary representation of protobuf message
+    val byteBuffer = ByteBuffer.allocate(5)
+
+    // Compression flag (1 byte) - 0, not compressed
+    byteBuffer.put(0)
+
+    // Message length
+    val messageLength = message.remaining()
```

Comment:
I think we need to do some conversion here to enforce that this number is big endian like we do here: https://github.com/lyft/Lyft-iOS/blob/f098c5aedfa27b0c0825aa779222c9596d0df511/Modules/LyftGRPC/Sources/GRPCStreamEmitter.swift#L41 (required by the gRPC spec)